plugins/techdocs/src/reader/components/TechDocsReaderPage/hooks.ts

```diff
@@ -44,3 +46,50 @@ export const useShadowRootElements = <
     .map(nodeList => Array.from(nodeList))
     .flat();
 };
+
+const isValidSelection = (newSelection: Selection) => {
+  // Safari sets the selection rect to top zero
+  return (
+    newSelection.toString() &&
+    newSelection.rangeCount &&
+    newSelection.getRangeAt(0).getBoundingClientRect().top
+  );
+};
+
+/**
+ * Hook for retreiving a selection within the ShadowRoot.
+ * @public
+ */
+export const useShadowRootSelection = (wait: number = 0) => {
```

Comment:
Just throwing an open thought here:

It would be possible to listen to the `mouseup` event once the first `selectionchange` event has been triggered, which means that no `debounce` would be needed.

A problem I see however is that non-pointer device selections such as keyboard or assisted screen readers would not fire the event. And then we would fallback to a debounce and it would be the same thing again.

Do you have any ideas if that would still be possible?